File: LAB 1/MATLAB/CMY-RGB.m
c_val = 0.70;
m_val = 0.90;
y_val = 0.10;

r_val = 1 - c_val;
g_val = 1 - m_val;
b_val = 1 - y_val;

fprintf('R = %.3f\n', r_val);
fprintf('G = %.3f\n', g_val);
fprintf('B = %.3f\n', b_val);

cmy_block = cat(3, c_val, m_val, y_val);
rgb_block = cat(3, r_val, g_val, b_val);

figure;
subplot(1,2,1), imshow(cmy_block), title('CMY Color');
subplot(1,2,2), imshow(rgb_block), title('Converted RGB Color');

File: LAB 1/Python/CMY-RGB.py
import numpy as np
import matplotlib.pyplot as plt

c_val = 0.70
m_val = 0.90
y_val = 0.10

r_val = 1 - c_val
g_val = 1 - m_val
b_val = 1 - y_val

print(f"R = {r_val:.3f}")
print(f"G = {g_val:.3f}")
print(f"B = {b_val:.3f}")

cmy_block = np.array([[[c_val, m_val, y_val]]])
rgb_block = np.array([[[r_val, g_val, b_val]]])

plt.figure()
plt.subplot(1, 2, 1)
plt.imshow(cmy_block)
plt.title("CMY Color")
plt.axis("off")

plt.subplot(1, 2, 2)
plt.imshow(rgb_block)
plt.title("Converted RGB Color")
plt.axis("off")

plt.show()
